detox/src/client/Client.js

```diff
@@ -73,16 +73,23 @@ class Client {
 
     try {
       if (this.isConnected) {
-        await this.sendAction(new actions.Cleanup(this._successfulTestRun));
+        this._scheduleAppTermination();
+
+        try {
+          await this.sendAction(new actions.Cleanup(this._successfulTestRun));
+        } catch (e) {
+          log.error({ event: 'ERROR' }, DetoxRuntimeError.format(e));
+        }
 
         this._whenAppIsConnected = this._invalidState('while cleaning up')
         this._whenAppIsReady = this._whenAppIsConnected;
       }
     } finally {
+      this._unscheduleAppTermination();
+      delete this.terminateApp; // property injection
+
       await this._asyncWebSocket.close();
```

Comment:
Probably, I should guard it here as well, to ensure the entire cleanup flow can pass. 🤔

Reply:
Both asyncwebsocket.close and sendAction are `.catch(() => log)`ged. We are good now.